client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,353 @@
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<RecurringPaymentsCouponAddEditModal { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'RecurringPaymentsCouponAddEditModal', () => {
+	let modalRoot;
+
+	const getRandomButton = () => screen.getByText( 'Random' );
+	const getCodeTextbox = () => screen.getByLabelText( 'Coupon code' );
+	const getDiscountTypeSelect = () => screen.getByLabelText( 'Discount type' );
+	const getAmountOption = () => screen.getByRole( 'option', { name: 'Amount' } );
+	const getPercentageOption = () => screen.getByRole( 'option', { name: 'Percentage' } );
+	const getAmountInput = () => screen.getByLabelText( 'Amount' );
+	const getLimitDurationToggle = () => screen.getByRole( 'checkbox', { name: 'Duration' } );
+	const getLimitDurationSelection = () =>
+		screen.getByRole( 'combobox', { name: 'duration selection' } );
+	const getLimitCouponsToggle = () =>
+		screen.getByRole( 'checkbox', { name: 'Limit coupon to specific emails' } );
+	const getLimitCouponsTextInput = () =>
+		screen.getByRole( 'textbox', { name: 'limit coupon to specific emails text input' } );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+		getconnectedAccountDefaultCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountDefaultCurrencyForSiteId
+		);
+		getconnectedAccountMinimumCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountMinimumCurrencyForSiteId
+		);
+
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		[ ...document.getElementsByClassName( 'ReactModalPortal' ) ].map( ( el ) => {
+			document.body.removeChild( el );
+		} );
+		modalRoot = null;
+		global.Date = RealDate;
+	} );
+
+	test( 'should render empty form when no coupon is provided', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const codeTextbox = getCodeTextbox();
+		const amountOption = getAmountOption();
+		const percentageOption = getPercentageOption();
+		const amountInput = getAmountInput();
+		const limitDurationToggle = getLimitDurationToggle();
+		const limitDurationSelection = getLimitDurationSelection();
+		const limitCouponsToggle = getLimitCouponsToggle();
+		const limitCouponsTextInput = getLimitCouponsTextInput();
+
+		expect( codeTextbox.value ).toEqual( '' );
+		expect( amountOption.selected ).toBe( false );
+		expect( percentageOption.selected ).toBe( true );
+		expect( amountInput ).name = 'discount_percentage';
+		expect( amountInput ).type = 'text';
+		expect( amountInput ).value = '';
```

Comment:
This is a bug since it's not actually an assertion - this actually assigns to the return object of `expect()`. Can we change to an actual assertion, like:

```suggestion
		expect( amountInput ).toHaveAttribute( 'name', 'discount_percentage' );
		expect( amountInput ).toHaveAttribute( 'type', 'text' );
		expect( amountInput ).not.toHaveValue();
```

This occurs multiple times in this PR.

Reply:
Yeah, I'm not sure what I was thinking here. I meant to write `expect( amount.name ).toEqual( 'discount_percentage' )`, etc.. Or at very least use `==` or `===` to get a failure!

What you've suggested is a much better option, I'll apply it throughout the PR.